localization/ekf_localizer/src/ekf_module.cpp

```diff
@@ -282,15 +283,37 @@ bool EKFModule::measurement_update_pose(
   return true;
 }
 
-geometry_msgs::msg::PoseWithCovarianceStamped EKFModule::compensate_pose_with_z_delay(
+geometry_msgs::msg::PoseWithCovarianceStamped EKFModule::compensate_rph_with_delay(
   const PoseWithCovariance & pose, const double delay_time)
 {
+  tf2::Quaternion delta_orientation;
+  if (last_angular_velocity_.length() > 0.0) {
+    delta_orientation.setRotation(
+      last_angular_velocity_.normalized(), last_angular_velocity_.length() * delay_time);
+  } else {
+    delta_orientation.setValue(0.0, 0.0, 0.0, 1.0);
+  }
+
+  tf2::Quaternion prew_orientation = tf2::Quaternion(
+    pose.pose.pose.orientation.x, pose.pose.pose.orientation.y, pose.pose.pose.orientation.z,
+    pose.pose.pose.orientation.w);
+
+  tf2::Quaternion curr_orientation;
+  curr_orientation = prew_orientation * delta_orientation;
+  curr_orientation.normalize();
+
   const auto rpy = autoware::universe_utils::getRPY(pose.pose.pose.orientation);
-  const double dz_delay = kalman_filter_.getXelement(IDX::VX) * delay_time * std::sin(-rpy.y);
-  PoseWithCovariance pose_with_z_delay;
-  pose_with_z_delay = pose;
-  pose_with_z_delay.pose.pose.position.z += dz_delay;
-  return pose_with_z_delay;
+  const double delta_z = kalman_filter_.getXelement(IDX::VX) * delay_time * std::sin(-rpy.y);
```

Comment:
Isn't this `rpy.y` the one not compensated? Maybe the rpy should be derived from `curr_orientation`.

Reply:
I didn't see any change in the test results. But theoretically I think it would be more accurate as you said. That's why I updated it. Thank you for your suggestion. @TaikiYamada4 
Related commit --> https://github.com/autowarefoundation/autoware.universe/pull/8095/commits/26f46c4ac9f5422b3adb6e326e281737e4ccd166